src/plugins/Url/index.js

```diff
@@ -0,0 +1,133 @@
+const Plugin = require('../../core/Plugin')
+const { h } = require('preact')
+const Provider = require('../Provider')
+// const Utils = require('../../core/Utils')
+require('whatwg-fetch')
+
+/**
+ * Link
+ *
+ */
+module.exports = class Link extends Plugin {
+  constructor (uppy, opts) {
+    super(uppy, opts)
+    this.id = this.opts.id || 'Url'
+    this.title = 'Link'
+    this.type = 'acquirer'
+    this.icon = () => <svg aria-hidden="true" class="UppyIcon UppyModalTab-icon" width="64" height="64" viewBox="0 0 64 64">
+      <circle cx="32" cy="32" r="31" />
+      <g fill-rule="nonzero" fill="#FFF">
+        <path d="M25.774 47.357a4.077 4.077 0 0 1-5.76 0L16.9 44.24a4.076 4.076 0 0 1 0-5.758l5.12-5.12-1.817-1.818-5.12 5.122a6.651 6.651 0 0 0 0 9.392l3.113 3.116a6.626 6.626 0 0 0 4.699 1.943c1.7 0 3.401-.649 4.697-1.943l10.241-10.243a6.591 6.591 0 0 0 1.947-4.696 6.599 6.599 0 0 0-1.947-4.696l-3.116-3.114-1.817 1.817 3.116 3.114a4.045 4.045 0 0 1 1.194 2.88 4.045 4.045 0 0 1-1.194 2.878L25.774 47.357z" />
+        <path d="M46.216 14.926a6.597 6.597 0 0 0-4.696-1.946h-.001a6.599 6.599 0 0 0-4.696 1.945L26.582 25.167a6.595 6.595 0 0 0-1.947 4.697 6.599 6.599 0 0 0 1.946 4.698l3.114 3.114 1.818-1.816-3.114-3.114a4.05 4.05 0 0 1-1.194-2.882c0-1.086.424-2.108 1.194-2.878L38.64 16.744a4.042 4.042 0 0 1 2.88-1.194c1.089 0 2.11.425 2.88 1.194l3.114 3.114a4.076 4.076 0 0 1 0 5.758l-5.12 5.12 1.818 1.817 5.12-5.122a6.649 6.649 0 0 0 0-9.393l-3.113-3.114-.003.002z" />
+      </g>
+    </svg>
+
+    // set default options
+    const defaultOptions = {}
+
+    this.opts = Object.assign({}, defaultOptions, opts)
+
+    this.hostname = this.opts.host
+
+    // Bind all event handlers for referencability
+    ;['getMeta', 'handleClick'].forEach(method => {
+      this[method] = this[method].bind(this)
+    })
+
+    this[this.id] = new Provider(uppy, {
+      host: this.opts.host,
+      provider: 'url',
+      authProvider: 'url'
+    })
+  }
+
+  getMeta (url) {
+    return fetch(`${this.hostname}/url/meta`, {
```

Comment:
based on my suggestion **2.** and **3.** above, we can have this `fetch` done in `ApiCalls`